### DISCUSSION TITLE
Struggle with verification of Taylor Factor calculations - Solution and further question

### DISCUSSION THREAD

**Discussion by Philip-Go** [mtex-toolbox/mtex#1997] [Tips and Tricks]
Hello everybody,

I recently started investigating my EBSD measurements on aluminium (fcc) regarding the Taylor Factor _**M**_ under destinct strain states.
However prior to analysis I had the issue with verifications of my code:
1. I wasn't able to recreate the value _**M**_ = 3.06 for a set of randomly distributed orientations unter uniaxial stress.
2. I as well couldn't find the relation of the Schmid Factor _m_ and Taylor Factor _**M**_ of _**M**_ = 1/ _**m**_ in the single crystaline case of a single cube-oriented (0,0,0) grain under uniaxial tensile loading in X direction, where _**m**_ = 0.408, thus _**M**_ = 2.45

I am no expert on this field and maybe these observation are clear to the experienced users in crystal plasticity, but maybe this can be of help for someone. However, this then leads me to a question at the bottom of the details, that I am not sure about and would be happy to get your opinion on it. 

TL;DR: the value of _**M**_ that is obtained from calcTaylor needs to be multiplied by the norm of the deformation tensor to fit the definition as σ/τ = **_M_** . Due to normalization of the strain tensor since MTEX 5.10, the Taylor Factor is too small in comparison to the literature. Further details follow below.

Details:
________________________________________________
Case 1
My code to calculate the Taylor Factor of randomly distributed orientations in aluminium:

setting up the sample orientations and the slip system:
`CS = crystalSymmetry('m-3m', [4 4 4], 'mineral', 'Aluminium', 'color', [0.53 0.81 0.98]);`
`sS = slipSystem.fcc(CS);`
`sS=sS.symmetrise;`
`oriRand =orientation.rand(5000,CS);`

defining the strain state for uniform elongation under uniaxial stress, like in a tensile test
`q=0.5`
`eps_uniaxStress = strainTensor(diag([-q -(1-q) 1]));`

calculating the mean Taylor Factor of that sample:
`[M,b,W] = calcTaylor(inv(oriRand)*eps_uniaxStress,sS);`
`mean(M)`
Due to the large set of random orientations this should yield a small deviation and result in `mean(M)`=2.50_ +/- 0.01

Obviously, this does not equal the much cited literature value of Taylors calculation (1938) of ~3.1 (http://pajarito.materials.cmu.edu/documents/Taylor_1938.pdf, http://pajarito.materials.cmu.edu/lectures/L11-PolyXtal_plast-Aniso3-24Feb14.pdf p.75).
Taking a look into the documentation of the VPSC (https://public.lanl.gov/lebenso/VPSC7c_manual.pdf) on page 102 gives the definition of the Taylor Factor. This leads to the conclusion, that the calculation of Taylor for his Taylor factor for a polycrystal under uniform elongation in uniaxial load is not normalized in regard to the magnitude of deformation.
Thus, the above calcuated `mean(M)` needs to be multiplied again with `norm(eps_uniaxStress)`:
`M_lit = mean(M) * norm(eps_uniaxStress)`
This now gives the Taylor Factor as known from literature: `M_lit` = 3.06 +/- 0.012

---------------------------------------------------
Case 2
Doing the same for only the cube orientation:
`cube = orientation.cube(CS);`
`[Mc,bc,Wc] = calcTaylor(inv(cube )*eps_uniaxStress,sS);`
`M_lit = mean(Mc) * norm(eps_uniaxStress)`
The resulting Taylor Factor is `M_lit` = 2.45

Now also calculating the Schmid Factor _**m**_:
`Stress_Ten = stressTensor.uniaxial(vector3d.Z)`;
`m = SchmidFactor(cube*sS,Stress_Ten)`;
This returns the Schmid Factors for all slip systems, showing that any slip system has either an absolute value of 0 (not active) or 0.408. This then leads to `1/m` = 1/0.408 = 2.45 = `M_lit`, so that the Taylor Factor and Schmid Factor can be transformed into oneanother for the sngle crystalline case.

_____________________________________________________

Remarks and further questions:
This leads me now to thequestion on your opinion, on which basis the Taylor Factor of different strain states can or should be discussed? I would suggest that, foremost, the norm of the strain state needs to be identical in the regarded strain states. Maybe someone can provide some literature to such cases, where Taylor Factors of different strain states are mentioned or even compared?
So far I am under the impression, that most readers of literature (and possibly some authors as well) are not always aware, that the Taylor Factor is sensitive to magnitude, much in contrast to the Schmid Factor, and think they can be easily transfered into one another - just as I was :D 

Looking forward to your comments or tips!
Cheers, Philip
